tests/contracts/root_chain/test_long_run.py

```diff
@@ -26,46 +26,61 @@ def test_slow(testlang):
     # 3. exit something
     # 4. attempt to finalize
 
-    for i in range(1000):
-        print("[{}]: iteration {}".format(datetime.datetime.now(), i))
+    # 10000 iterations correspond to 3 days of running time
+    # To make test run for 16 hours (i.e. "overnight") set number of iterations to 2000
+    for i in range(2000):
+        print("[{}]: iteration {}; size {}".format(datetime.datetime.now(), i, len(utxos)))
         # 1. deposit few
-        for _ in range(5):
+        for _ in range(6):
+            testlang.ethtester.chain.mine()
             deposit_id = testlang.deposit(owner.address, amount)
             max_gas = max(max_gas, testlang.ethtester.chain.last_gas_used())
+            testlang.ethtester.chain.mine()
             spend_id = testlang.spend_utxo(deposit_id, owner, amount, owner)
             max_gas = max(max_gas, testlang.ethtester.chain.last_gas_used())
             utxos.append(spend_id)
 
         testlang.forward_timestamp(DAY * 2)
 
-        testlang.ethtester.chain.mine()
+        print("done depositing new")
         # 2. spend
         for _ in range(random.randint(2, 4)):
+            testlang.ethtester.chain.mine()
             utxos, pos = pick(utxos)
             spend_id = testlang.spend_utxo(pos, owner, amount, owner)
             max_gas = max(max_gas, testlang.ethtester.chain.last_gas_used())
             utxos.append(spend_id)
 
+        print("done spending")
+
         # 3. double-spend, exit and challenge
         for _ in range(random.randint(2, 4)):
+            testlang.ethtester.chain.mine()
             utxos, pos = pick(utxos)
             spend_id = testlang.spend_utxo(pos, owner, amount, owner)
             max_gas = max(max_gas, testlang.ethtester.chain.last_gas_used())
             utxos.append(spend_id)
+            testlang.ethtester.chain.mine()
             testlang.start_standard_exit(owner, pos)
+            max_gas = max(max_gas, testlang.ethtester.chain.last_gas_used())
+            testlang.ethtester.chain.mine()
             testlang.challenge_standard_exit(pos, spend_id)
+            max_gas = max(max_gas, testlang.ethtester.chain.last_gas_used())
 
-        testlang.ethtester.chain.mine()
+        print("done double-spending")
         # 4. exit
         for _ in range(random.randint(2, 4)):
+            testlang.ethtester.chain.mine()
             utxos, pos = pick(utxos)
             testlang.start_standard_exit(owner, pos)
             max_gas = max(max_gas, testlang.ethtester.chain.last_gas_used())
 
+        print("done exiting")
         testlang.ethtester.chain.mine()
         # 4. attempt to finalize
         testlang.finalize_exits(NULL_ADDRESS, 0, 3)
         max_gas = max(max_gas, testlang.ethtester.chain.last_gas_used())
         print("max_gas is {}".format(max_gas))
+        print("done finalizing")
 
     assert max_gas < 200000
```

Comment:
maybe we should track `max_gas` for every contract function separately, assert separately and have tighter limit. So that excessive gas usage in a cheaper kind of transaction doesn't get silenced by `200000` being the limit suitable for an expensive one.